tokio-stream/src/wrappers/mpsc_unbounded.rs

```diff
@@ -61,6 +61,22 @@ impl<T> Stream for UnboundedReceiverStream<T> {
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         self.inner.poll_recv(cx)
     }
+
+    /// Returns the bounds of the stream based on the underlying receiver.
+    ///
+    /// For open channels, it returns `(receiver.len(), None)`.
+    ///
+    /// For closed channels, it returns `(receiver.len(), receiver.len())`.
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.inner.len();
+        let upper = if self.inner.is_closed() {
+            Some(len)
+        } else {
+            None
+        };
```

Comment:
Consider this scenario:

1. We read `self.inner.len()`.
2. Another thread sends a message.
3. Another thread drops the last sender.
4. We read `self.inner.is_closed()` and get true.

And now the value we got from `len` is incorrect. Please call `is_closed()` before calling `len()`. This ensures that dropping the last sender happens-before the call to `len()`.